service/history/execution/context.go

```diff
@@ -1215,6 +1217,20 @@ func (c *contextImpl) updateWorkflowExecutionWithRetry(
 		return err
 	}
 
+	//Calculating the TTL for workflow Execution.
```

Comment:
The following logic is better to be moved to `CloseTransactionAsMutation` function.
And it can be shared with UpdateWorkflow and ConflictResolveWorkflow.